src/components/House/House.tsx

```diff
@@ -1,53 +1,121 @@
+        }
+
+        useEffect(() => {
+
+                if (queue.length !== 0 && queue[0] !== undefined) {
+
+                    const differenceR = Math.abs(queue[0] - targetLiftRight)
+                    const differenceL = Math.abs(queue[0] - targetLiftLeft)
+
+                    // удаляет дубляж целей в очереди
+                    if (queue[0] === targetLiftRight || queue[0] === targetLiftLeft) {
+                        setQueue([...queue].slice(1))
+                        setLiftInMotion((prev) => ({...prev, rightLift: false, leftLift: false}))
+                    }
+
+                    // если едет только 1 лифт, тогда запускаем второй
+                    else if ((liftInMotion.leftLift && !liftInMotion.rightLift) ||
+                        (!liftInMotion.leftLift && liftInMotion.rightLift)) {
 
-    const onStopLift = () => {
-        setFloorValueButton(
-            {...floorValueButton, isPressed: false}
+                        if (!liftInMotion.rightLift) {
+                            setLiftInMotion((prev) => ({...prev, rightLift: true}))
+                            setTargetLiftRight(queue[0])
+                            setQueue([...queue].slice(1))
+
+                        } else if (!liftInMotion.leftLift) {
+                            setLiftInMotion((prev) => ({...prev, leftLift: true}))
+                            setTargetLiftLeft(queue[0])
+                            setQueue([...queue].slice(1))
+                        }
+                    }
+
+                    // если оба стоят, то условие с дистанцией едет, который ближе
+                    else if (!liftInMotion.leftLift && !liftInMotion.rightLift) {
+
+                        if (differenceR <= differenceL) {
+                            setLiftInMotion((prev) => ({...prev, rightLift: true}))
+                            setTargetLiftRight(queue[0])
+                            setQueue([...queue].slice(1))
+                        } else {
+                            setLiftInMotion((prev) => ({...prev, leftLift: true}))
+                            setTargetLiftLeft(queue[0])
+                            setQueue([...queue].slice(1))
+                        }
+                    }
+                }
+            }, [liftInMotion, targetLiftLeft, targetLiftRight, queue]
         )
+
+        const onStopLift = (liftLocation: string | undefined) => {
+            if (liftLocation !== undefined) {
+                liftLocation === 'left' ?
+                    setLiftInMotion((prev) => ({...prev, leftLift: false})) :
+                    setLiftInMotion((prev) => ({...prev, rightLift: false}))
+            }
+        }
+
+        return (
+            <SHouse>
+                <Container>
+                    {floors.map((f) => (
+                            <Floor key={f.id}
+                                   floor={f.floor}
+                                   addQueue={addQueue}
```

Comment:
Чтобы не пришлось передавать и так известный параметр

```suggestion
                                   addQueue={()=>addQueue(f.floor)}
```